core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -234,6 +315,34 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
   }
 
   /**
+   * Sets the topic znode with the given assignment.
+   * @param topic the topic whose assignment is being set.
+   * @param assignment the partition to replica mapping to set for the given topic
+   * @throws KeeperException if there is an error while setting assignment
+   */
+  def setTopicAssignment(topic: String, assignment: Map[TopicPartition, Seq[Int]]) = {
+    val setDataResponse = setTopicAssignmentRaw(topic, assignment)
+    if (setDataResponse.resultCode != Code.OK){
+      setDataResponse.resultException.foreach(e => throw e)
+    }
+  }
+
+  /**
+   * Create the topic znode with the given assignment.
+   * @param topic the topic whose assignment is being set.
+   * @param assignment the partition to replica mapping to set for the given topic
+   * @throws KeeperException if there is an error while creating assignment
+   */
+  def createTopicAssignment(topic: String, assignment: Map[TopicPartition, Seq[Int]]) = {
+    createRecursive(TopicsZNode.path)
+    val createRequest = CreateRequest(TopicZNode.path(topic), TopicZNode.encode(assignment), acls(TopicZNode.path(topic)), CreateMode.PERSISTENT)
+    val createResponse = retryRequestUntilConnected(createRequest)
+    if (createResponse.resultCode != Code.OK){
+      createResponse.resultException.foreach(e => throw e)
+    }
+  }
```

Comment:
Could this method be just  createRecursive(TopicZNode.path(topic), TopicZNode.encode(assignment))?